src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -3063,6 +3063,75 @@ static void F(object x)
                 Diagnostic(ErrorCode.WRN_NullReferenceArgument, "y").WithArguments("x", "void Program.F(object x)").WithLocation(11, 11));
         }
 
+        [Fact]
+        public void Directive_GloballyEnabled_GeneratedCode_Ignored()
+        {
+            var source =
+@"
+// <autogenerated />
+class Program
+{
+    static void F(object x)
+    {
+        x = null; // no warning, generate file
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugDll.WithNullableContextOptions(NullableContextOptions.Enable));
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Directive_GloballyEnabled_GeneratedCode_Enabled_Disabled()
+        {
+            var source =
+@"
+// <autogenerated />
+class Program
+{
+#nullable enable
+    static void F(object x, object y)
+    {
+        x = null; // warning
+#nullable disable
+        y = null; // no warning
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugDll.WithNullableContextOptions(NullableContextOptions.Enable));
+            comp.VerifyDiagnostics(
+                 // (8,13): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                 //         x = null; // warning
+                 Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(8, 13)
+                );
+        }
+
+
+        [Fact]
+        public void Directive_GloballyEnabled_GeneratedCode_Enabled_Restore()
+        {
+            var source =
+@"
+// <autogenerated />
+class Program
+{
+#nullable enable
+    static void F(object x)
+    {
+        x = null; // warning
+    }
+#nullable restore
+    static void G(object y)
```

Comment:
>static void G(object y) [](start = 4, length = 23)

Consider merging with method above so it's clear the only difference from the preceding test is the `#nullable restore`.

Reply:
Actually, I realize that would make `object y` null-oblivious. Given that, perhaps change the other tests to match this one, with two methods in each.

---
In reply to: [253182819](https://github.com/dotnet/roslyn/pull/33048#discussion_r253182819) [](ancestors = 253182819)